blog/_posts/2023-03-12-progress-report.markdown

```diff
@@ -0,0 +1,62 @@
+---
+layout: post
+title:  "First post, progress report!"
+date:   2023-03-12 01:23:00 +0100
+categories:
+---
+Get ready for the biggest Ruffle announcement yet! And the first one on the blog!
+
+# Huge improvements to Ruffle's AVM1 engine accuracy!
+Thanks to a massive code refactor by @CUB3D, dozens upon dozens of ActionScript 2 games have been fixed! Here are just a few of them:
+- Chibi Knight: [https://www.newgrounds.com/portal/view/526470](https://www.newgrounds.com/portal/view/526470)
+- Xeno Tactic 2: [https://www.newgrounds.com/portal/view/438241/format/flash?emulate=flash](https://www.newgrounds.com/portal/view/438241/format/flash?emulate=flash)
+- Trojan War: [https://www.newgrounds.com/portal/view/604949/format/flash?emulate=flash](https://www.newgrounds.com/portal/view/604949/format/flash?emulate=flash)
+- Frontline Defense: [https://www.kongregate.com/games/kazama_bee/frontline-defense-first-assault](https://www.kongregate.com/games/kazama_bee/frontline-defense-first-assault)
+- Bubble Bobble: The Revival: [https://flasharch.com/en/archive/play/084db17094dc7452f01da0905d7eb516](https://flasharch.com/en/archive/play/084db17094dc7452f01da0905d7eb516)
+- Cube Colossus: [https://www.newgrounds.com/portal/view/507205/format/flash?emulate=flash](https://www.newgrounds.com/portal/view/507205/format/flash?emulate=flash)
+- Chronotron: [https://www.kongregate.com/games/Scarybug/chronotron](https://www.kongregate.com/games/Scarybug/chronotron)
+- The Powerpuff Girls: Attack of the Puppybots: [https://flasharch.com/en/archive/play/b93354279e9788b849f83ef78f52cbbb](https://flasharch.com/en/archive/play/b93354279e9788b849f83ef78f52cbbb)
+- Extreme Pamplona: [https://flasharch.com/en/archive/play/e36aac73914ec8672218317e000615d7](https://flasharch.com/en/archive/play/e36aac73914ec8672218317e000615d7)
+
+# Incredible progress in AVM2 (ActionScript 3) support!
+- Our website now has a page listing exactly what ActionScript 3 APIs we have implemented, making it easy to follow our progress. It will be frequently updated! [https://ruffle.rs/avm2.html](https://ruffle.rs/avm2.html)
+- XML support is rapidly improving! ActionScript 3 games tend to use a wide variety of XML methods. As Ruffle gains support for these methods, games are springing to life!
+- Several problems that caused unresponsive buttons and menus have been fixed. Unclickable buttons in ActionScript 3 games are (mostly) a thing of the past!
+- Other major refactors are in progress to improve Ruffle's compatibility with ActionScript 3 frameworks like Haxe.
+- @Aaron1011 is working on **Stage3D support**! Thanks to his work, the gorgeous interstellar strategy game Solarmax 2 will soon be playable in Ruffle! Check out the video below.

```

Comment:
Since this was written, Solarmax2 became fully playable (at least on the desktop player).

Reply:
Similarly, dynamic audio buffering is out now. So yeah, there's a few things that need to be rephrased

Reply:
How's that?